src/client/clientmap.cpp

```diff
@@ -1197,13 +1197,22 @@ void ClientMap::renderMapShadows(video::IVideoDriver *driver,
 	// Render all mesh buffers in order
 	drawcall_count += draw_order.size();
 
+	bool translucent_foliage = g_settings->getBool("enable_translucent_foliage");
+
+	video::E_MATERIAL_TYPE leaves_material = video::EMT_SOLID;
+
+	// For translucent leaves, we want to use backface culling instead of frontface.
+	if (translucent_foliage) {
+		const u32 leaves_shader = m_client->getShaderSource()->getShader("nodes_shader", TILE_MATERIAL_WAVING_LEAVES, NDT_ALLFACES);
+		leaves_material = m_client->getShaderSource()->getShaderInfo(leaves_shader).material;
+	}
```

Comment:
I don't have a good suggestion but this looks like totally breaking the abstraction.
Maybe it could be read from a `TileLayer` of an existing leaf node (`NodeDefManager`)?

Reply:
This one I find hard to fix... This is something that isn't so much inherent to the material as it is the shader really, so breaking the abstraction is kind of an inherent thing to the problem unfortunately. I'm open to suggestions but I have no idea how to improve it tbh...